### PR TITLE
Library cleanup part 1

### PR DESCRIPTION
**Pull request by signedav**
I found and solved the following incompatibilities with version 6.4.3 considering **_libili2db_** and _**libprojectgen**_.

### libili2db:

Info: Introduced `def metaconfig_directories(self): ` does not hurt. 

Info: `IliCache` received some generalization but works the same way than before. New Classes like `IliDataCache` where introduced, but without any impacts.

Info: In the `ili2dbconfig` there is the part with the `metaconfig` implemented. If no `metaconfig` is passed to the config, then it works like before. It sets the parameters that generally make sense. The ones defined as variables can be overridden.

Resolved conflict: in the `ili2dbconfig`  we set the default value `self.create_basket_col = True`
The default parameters can be overwritten from outside, so I set it back to `False`

Info: in the `ili2dbconfig` we set the default value `self.srs_code = 2056` from `21781`, what should not hurt anyone.

Unresolved Conflict: `ValidateConfiguration` and` ilivalidator.py `that should be melted with @gacarrillor s implementation.

### libqgsprojectgen

Resolved conflict: `def __init__(self, name=None, expanded=True, ignore_node_names=None):`
`expanded` was removed. This because parameters like  `checked`, `mutually_exclusive` can be set with the new version, so expanded should work the same way. I reintroduced it for the backwards compatibility.

Info: In the legend function of the generator it's possible to pass a `layertree` structure. But this does not influence the existing solution.

Info: in class Layer we can pass the parameters and act accordingly, this does not hurt as long as they are not set automatically.
`is_basket_table=False, is_dataset_table = False, model_topic_name=None)` but see below.

Resolved conflict: We removed the `BASKET_TABLES` from the `IGNORED_TABLES`. 
And we put em back to ignored tables and handle them as exceptions. For this a parameter is introduced `ignore_basket_tables` default `True`.

Resolved conflict: stuff is made according to model_topic_name and this is generated  for every table...

Resolved conflict: `Generator` and `DBConnector` basket_handling behavior is used according to `basket_handling = self.get_basket_handling()`
I thought about creating a `class BasketGenerator(Generator)` but this is lot's of duplicate code only by overriding the `def layers`.
So the straight forward solution is to pass the `consider_basket_handling` parameter in the constructor. If not passed even tables named as basket tables are not handled as basket tables (as normal tables instead) but usually they are ignored anyway. `model_topic_name` is generated anyway and written to the layer properties. I think it's a useful info anyway.

### CODE REVIEW

**Comment by signedav** [comment]
@gacarrillor Would be great if you could have a look if it would work for you like this.
Next steps will be that I fix up the structure to have an independent library.

**Comment by signedav** [comment]
(Test fixes in the PR #641 with the Model Baker Plugin part)

**Comment by signedav** [comment]
Some more changes I did:
- put the utils where they needed. `qt_utils` will be in a utils folder in the library. `qgis_utils` in a subdir of the libqgsprojectgen part, the other utils will stay out of the library
- made ili2db part more independent (like you @gacarrillor did in your release): introduced ili2btools / ili2dbargs and cleaned up globals (moved `DbActionType` to libqgsprojectgen and `displayDbIliMode` to the plugin globals).
- used the version check function from Asistente not to need to have the `packaging `package
- passing the global parameters of the plugin (`CATALOGUE_DATASETNAME`) via context to the `Project`

With this changes libili2db works at it's own (needing utils.qt_utils). libqgsprojectgen needs libili2db (and utils.qt_utils) and the plugin needs them both.

File: QgisModelBaker/libqgsprojectgen/generator/generator.py [hunk 1]
```diff
@@ -129,10 +132,11 @@ def layers(self, filter_layer_list=[]):
             is_attribute = bool(record.get("attribute_name"))
             is_structure = record.get("kind_settings") == "STRUCTURE"
             is_nmrel = record.get("kind_settings") == "ASSOCIATION"
-            is_basket_table = (
+            # when the basket handling is not considered we do not consider tables named as basket tables consider as basket tables
```
Comment (gacarrillor) [inline QgisModelBaker/libqgsprojectgen/generator/generator.py:135]: Is this right? The double `consider` makes it difficult to understand.

**Review by gacarrillor — COMMENTED** [review]
+ [x] First review. Minor comments added inline.

Other questions (if this PR is WIP, that would already answer my first 2 questions :) ):

 + Where do you have iliexporter/importer/updater/validator modules?
 + Where do you have fields/layers/form/relations modules?

 + Regarding one of the inline comments, we now use a `group` parameter for creating the project. See this commit: https://github.com/SwissTierrasColombia/QgisModelBaker/commit/31cca21d8838bbfa9044673c7c5da0f2d0a010d8 We can discuss about it if it's unclear.

 + Don't fully get this info from above: 

    > Resolved conflict: stuff is made according to model_topic_name and this is generated for every table...

    Do you mean that now MB gets the `model_topic_name` for each table because it uses it for something else? If so, yes, I think it shouldn't be a problem.

**Comment by signedav** [comment]
Thanks @gacarrillor 

> Where do you have iliexporter/importer/updater/validator modules?
> Where do you have fields/layers/form/relations modules?

It's not WIP but part 1. In this PR I only made it independent from the plugin. If modules are not in this PR they won't be changed. I make it step by step, so it's better reviewable.
After this is merged, I will create a new structure:

```
- package: modelbaker
├─ iliwrapper
   ├─ globals.py
   ├─ ili2dbargs.py
   ├─ ili2dbconfig.py
   ├─ [...]
├─ qgismaker
   ├─ dataobjects
   │  ├─ fields.py
   │  ├─ form.py
   │  ├─ [...]
   ├─ dbfactory
   │  ├─ db_command_config_manager.py
   │  ├─ db_factory.py
   │  ├─ [...]
   ├─ dbconnector
   │  ├─ [...]
   ├─ generator
   │  ├─ [...]
   ├─ utils
   │  ├─ globals.py
   │  ├─ qgis_utils.py
   ├─ [...]
├─ utils
   ├─ qt_utils.py
```

Concerning the names `ìliwrapper` (ex. `libili2db) and `qgismaker` (ex. `libqgsprojectgen`) I tried to be (a) short, (b) avoid "lib", (c) avoid the thought that it could be a library from ili2db or from qgis itself and (d) avoid short forms. With `iliwrapper` I'm quite happy, with `qgismaker` not that much, but still better than a long `qgisprojectgenerator` or just `qgisgenerator` what would lead to import `qgisgenerator.generator.generator` etc... Open for inputs.

> Resolved conflict: stuff is made according to model_topic_name and this is generated for every table...

I noted, that the modules in libqgsprojectgen do things when `model_topic_name` is set. But it's only the setting of the layer variable and the setting of the filter expression in the relation reference created when the table is a basket table. So this does not change the behavior in other plugins.

I will have a look at the group implementation.

File: QgisModelBaker/libqgsprojectgen/generator/generator.py (hunk 1 above)
Comment (signedav) [reply QgisModelBaker/libqgsprojectgen/generator/generator.py:135]: Even confusing for me now. But it's correct. I will change the comment.

File: QgisModelBaker/libili2db/ili2dbutils.py [hunk 2]
```diff
@@ -253,15 +251,71 @@ def getenv_system(varname, default=""):
...
 
 
+def is_version_valid(
+    current_version,
+    min_required_version,
+    exact_required_version=False,
+    module_tested="",
+):
+    """
+    Generic one, it helps us to validate whether a current version is greater or equal to a min_required_version or,
```
[the remark this answers is not in the document]
Comment (signedav) [reply QgisModelBaker/libili2db/ili2dbutils.py:274]: Btw. Happy how I solved it in ili2dbargs.py and ili2dbtools.py?

File: QgisModelBaker/libqgsprojectgen/dataobjects/project.py [hunk 3]
```diff
@@ -25,15 +25,14 @@
 from QgisModelBaker.libqgsprojectgen.dataobjects.layers import Layer
 from QgisModelBaker.libqgsprojectgen.dataobjects.legend import LegendGroup
 from QgisModelBaker.libqgsprojectgen.dataobjects.relations import Relation
-from QgisModelBaker.utils.globals import CATALOGUE_DATASETNAME
 
 ENUM_THIS_CLASS_COLUMN = "thisclass"
 
 
 class Project(QObject):
```
[the remark this answers is not in the document]
Comment (signedav) [reply QgisModelBaker/libqgsprojectgen/dataobjects/project.py:37]: I integrated the group parameter now.

**Comment by signedav** [comment]
Merging this - thanks @gacarrillor

**Pull request merged**